OpenTrons/opentrons-api
# fix(app): set max height for desktop modal shell

closes [RQA-2279](https://opentrons.atlassian.net/browse/RQA-2279)

# Overview

We special-cased the size of the modal shell to accommodate a large body for the 96-channel pipette attachment BeforeBeginning step to avoid scrollable content. The size was set to 70% of the screen, but that resulted in a lot of modal whitespace when the app was resized to larger than default dimensions. Here, I add a max height of 31rem to accommodate the content of the largest pipette wizard flow tile content without growing arbitrarily large with a resized desktop app. I also ensure minimum padding between tile content and footer buttons, and fix padding on the ChoosePipette tile.

# Test Plan

- launch 96-channel attachment wizard flow
- resize desktop app to larger than default in the vertical direction
- observe that modal shell does not resize to larger than the content

I also smoketested various wizard flows that use LegacyModalShell.

<img width="1728" alt="Screenshot 2024-04-16 at 10 11 31 AM" src="https://github.com/Opentrons/opentrons/assets/47604184/b30ca3c3-22e4-49a4-b2af-593c3a463d84">

# Changelog

- set special-cased LegacyModalShell height to constant rather than percentage to not grow with resized app
- add padding between body and bottom buttons of GenericWizardTile
- set explicit max height to 31rem for LegacyModalShell if not full screen
- fix container padding for ChoosePipette

# Review requests

frontend friends

# Risk assessment

low

[RQA-2279]: https://opentrons.atlassian.net/browse/RQA-2279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
